internal/manifest/parse.go

```diff
@@ -115,3 +146,60 @@ func pullLayer(ctx context.Context, imageRef string, keyChain authn.Keychain) (c
 func getFsChartPath(imageSpec v1beta2.ImageSpec) string {
 	return filepath.Join(os.TempDir(), fmt.Sprintf("%s-%s", imageSpec.Name, imageSpec.Ref))
 }
+
+func untarLayer(tarPath string) (string, error) {
+	tarFile, err := os.Open(tarPath)
+	if err != nil {
+		return "", fmt.Errorf("failed to open file: %w", err)
+	}
+	defer tarFile.Close()
+
+	tarReader := tar.NewReader(tarFile)
+
+	for {
+		header, err := tarReader.Next()
+		if errors.Is(err, io.EOF) {
+			break
+		}
+		if err != nil {
+			return "", fmt.Errorf("failed to read tar: %w", err)
+		}
+
+		if strings.HasPrefix(header.Name, "._") {
+			continue
+		}
+
+		extractedFilePath, err := sanitizeArchivePath(filepath.Dir(tarPath), header.Name)
+		if err != nil {
+			return "", fmt.Errorf("failed to sanitize archive path: %w", err)
+		}
+
+		if _, err := os.Stat(extractedFilePath); err == nil {
+			return extractedFilePath, nil
+		}
+
+		outFile, err := os.Create(extractedFilePath)
+		if err != nil {
+			return "", fmt.Errorf("failed to create extracted file: %w", err)
+		}
+		defer outFile.Close()
+
+		var maxBytes int64 = 1024 * 1024
+		if _, err := io.CopyN(outFile, tarReader, maxBytes); err != nil && !errors.Is(err, io.EOF) {
```

Comment:
why not use `io.Copy`?

Reply:
That is what I did first but a linter had a problem with this - since zip bombs exist, a file might have a small compressed size but huge size when uncompressed so the bytes must be limited

Reply:
but CopyN only copy the maxBytes, you can't make sure the file get fully copied.